test/testother.cpp

```diff
@@ -1732,6 +1732,19 @@ class TestOther : public TestFixture {
                       "[test.cpp:4]: (style) The scope of the variable 'b' can be reduced.\n"
                       "[test.cpp:5]: (style) The scope of the variable 'c' can be reduced.\n",
                       errout_str());
+
+        check("struct S { int a; };\n" // #12618
+              "int f(const S* s, int i) {\n"
+              "    int x = s->a;\n"
+              "    const int b[] = { 1, 2, 3 };\n"
```

Comment:
Even the scope of 'b' could be reduced.

Reply:
yeah.. but that requires that scope of y can be reduced. I don't know if we will warn about b if y is moved.

Reply:
We don't warn for arrays because of
~~~c++
if (Token::Match(tok, "= %varid%", var->declarationId()) && (var->isArray() || var->isPointer() || (var->valueType() && var->valueType()->container))) // Create a copy of array/pointer. Bailout, because the memory it points to might be necessary in outer scope
            return false;
~~~

Reply:
ok it's a bailout we should fix.. good that the comment says "bailout" explicitly so we can search for it..